File: ABC/ABC333/D/Main.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class Main {
  public static int countSubtreeSize(ArrayList<ArrayList<Integer> > graph, int v, int parent) {
    int ret = 1;
    for (int i = 0; i < graph.get(v).size(); ++i) {
      int nv = graph.get(v).get(i);
      if (nv != parent) {
        ret += countSubtreeSize(graph, nv, v);
      }
    }
    return ret;
  }
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int n = Integer.parseInt(sc.next());
    ArrayList<ArrayList<Integer> > graph = new ArrayList<>();
    for (int i = 0; i < n; ++i) {
      ArrayList<Integer> g = new ArrayList<>();
      graph.add(g);
    }
    for (int i = 0; i < n - 1; ++i) {
      int u = Integer.parseInt(sc.next());
      int v = Integer.parseInt(sc.next());
      graph.get(--u).add(--v);
      graph.get(v).add(u);
    }
    sc.close();

    ArrayList<Integer> subtreeSizes = new ArrayList<>();
    for (int i = 0; i < graph.get(0).size(); ++i) {
      subtreeSizes.add(countSubtreeSize(graph, graph.get(0).get(i), 0));
    }
    Collections.sort(subtreeSizes);

    int res = 1;
    for (int i = 0; i < subtreeSizes.size() - 1; ++i) {
      res += subtreeSizes.get(i);
    }
    System.out.println(res);
  }
}

File: ABC/ABC333/D/main.cc
#include <algorithm>
#include <iostream>
#include <numeric>
#include <vector>

int SubtreeSize(const std::vector<std::vector<int> > &graph, const int &v, const int &parent) {
  if ((int)graph[v].size() == 1 && graph[v][0] == parent) return 1;
  int ret = 1;
  for (int child : graph[v]) {
    if (child != parent) ret += SubtreeSize(graph, child, v);
  }
  return ret;
}

int main() {
  int n;
  std::cin >> n;

  std::vector<std::vector<int> > graph(n);
  for (int i = 0; i < n - 1; ++i) {
    int u, v;
    std::cin >> u >> v;
    graph[--u].emplace_back(--v);
    graph[v].emplace_back(u);
  }

  std::vector<int> subtree_sizes;
  for (int child : graph[0]) {
    subtree_sizes.emplace_back(SubtreeSize(graph, child, 0));
  }
  std::sort(subtree_sizes.begin(), subtree_sizes.end());
  int res = std::accumulate(subtree_sizes.begin(), std::prev(subtree_sizes.end()), 1);
  std::cout << res << std::endl;
  return 0;
}